floodlight/io/dfl.py

```diff
@@ -0,0 +1,324 @@
+    pitch = _read_pitch_from_mat_info(filepath_mat_info)
+
+    # create or check links
+    if links is None or id_to_jrsy is None:
+        links, id_to_jrsy = create_links_from_mat_info(filepath_mat_info)
+    else:
+        pass
+        # potential check
+
+    # create periods
+    periods, framerate = _create_periods_from_dat(filepath_dat)
+    segments = list(periods.keys())
+
+    # infer data array shapes
+    number_of_home_players = max(links["Home"].values())
+    number_of_away_players = max(links["Away"].values())
+    number_of_frames = {}
+    for segment in segments:
+        start = periods[segment][0]
+        end = periods[segment][1]
+        number_of_frames[segment] = end - start + 1
+
+    # bins
+    xydata = {
+        "Home": {
+            segment: np.full(
+                [number_of_frames[segment], number_of_home_players * 2], np.nan
+            )
+            for segment in segments
+        },
+        "Away": {
+            segment: np.full(
+                [number_of_frames[segment], number_of_away_players * 2], np.nan
+            )
+            for segment in segments
+        },
+        "Ball": {
+            segment: np.full([number_of_frames[segment], 2], np.nan)
+            for segment in segments
+        },
+    }
+    codes = {
+        code: {segment: [] for segment in segments}
+        for code in ["possession", "ballstatus"]
+    }
+
+    # loop over frame sets containing player & ball positions for all segments
+    for _, frame_set in etree.iterparse(filepath_dat, tag="FrameSet"):
+
+        # ball
+        if frame_set.get("TeamId") == "Ball":
+            # (x, y) position
+            segment = frame_set.get("GameSection")
+            xydata["Ball"][segment][:, 0] = np.array(
+                [float(frame.get("X")) for frame in frame_set.iterfind("Frame")]
+            )
+            xydata["Ball"][segment][:, 1] = np.array(
+                [float(frame.get("Y")) for frame in frame_set.iterfind("Frame")]
+            )
+            # codes
+            codes["ballstatus"][segment] = [
+                float(frame.get("BallStatus")) for frame in frame_set.iterfind("Frame")
+            ]
+            codes["possession"][segment] = [
+                float(frame.get("BallPossession"))
+                for frame in frame_set.iterfind("Frame")
+            ]
+
+        # teams
+        else:
+            # find identity of frame set
+            frames = [frame for frame in frame_set.iterfind("Frame")]
+            segment = frame_set.get("GameSection")
+            if frame_set.get("PersonId") in id_to_jrsy["Home"]:
+                team = "Home"
+                jrsy = id_to_jrsy[team][frame_set.get("PersonId")]
+            elif frame_set.get("PersonId") in id_to_jrsy["Away"]:
+                team = "Away"
+                jrsy = id_to_jrsy[team][frame_set.get("PersonId")]
+            else:
+                team = None
+                jrsy = None
+                pass
```

Comment:
```suggestion
```

Reply:
This was added to resolve a warning in the IDE, is now removed.